firebase.go

```diff
@@ -230,6 +230,19 @@ func (fb *Firebase) doRequest(method string, body []byte) ([]byte, error) {
 	}
 
 	resp, err := fb.client.Do(req)
+
+	// 307 handling.
+	// replay the previous request using the new url in the Location header
+	// Ref :  https://code.google.com/p/go/issues/detail?id=7912
+	if strings.HasPrefix(resp.Status, "307 ") {
```

Comment:
Can you use the `resp.StatusCode` variable instead and compare against the http package constant?

```go
if resp.StatusCode == http.StatusTemporaryRedirect {

}
```